dev/com.ibm.ws.jdbc/src/com/ibm/ws/rsadapter/impl/PostgreSQLHelper.java

```diff
@@ -31,23 +31,52 @@
     
     private static final String PROP_DEFAULT_FETCH_SIZE = "DefaultFetchSize";
     private static final String PROP_PREPARE_THRESHOLD = "PrepareThreshold";
+    private static final String PROP_AUTOSAVE = "AutoSave";
 
     @SuppressWarnings("deprecation")
     private transient com.ibm.ejs.ras.TraceComponent jdbcTC = com.ibm.ejs.ras.Tr.register("com.ibm.ws.postgresql.logwriter", "WAS.database", null);
 
     private transient PrintWriter jdbcTraceWriter;
+    
+    private Class<?> Autosave;
+    private Class<?> LargeObjectManager;
+    private Class<?> BaseConnection;
 
-    PostgreSQLHelper(WSManagedConnectionFactoryImpl mcf) {
+    PostgreSQLHelper(WSManagedConnectionFactoryImpl mcf) throws ClassNotFoundException {
         super(mcf);
     }
+    
+    /**
+     *  Proxy method for org.postgresql.PGConnection.getLargeObjectAPI()
+     *  During 19.0.0.7 this method was blocked because it closed the underlying connection on some code paths.
+     *  However, we later realized that by overriding this method and constructing the LargeObjectManager using
+     *  a proxied instance of this class, we can maintain control of the connection that the LargeObjectAPI uses.
+     */
+    public Object getLargeObjectAPI(Connection con) throws SQLException {
```

Comment:
The postgres doc says "For Postgres, you must access Large Objects within an SQL transaction. ", can we enforce that as part of our proxying.?

Reply:
The implementation of PostgreSQL enforces this on its own. For example, in `LargeObjectManager.createLO()` it verifies that `autoCommit == false` and blows up otherwise